aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -438,70 +424,50 @@ public void observe(
      * @param onComplete Invoked if the call is successful.
      * @param onError Invoked if not successful
      */
+    @SuppressWarnings("unused")
     @Override
     public void clear(@NonNull Action onComplete,
                       @NonNull Consumer<DataStoreException> onError) {
         // We shouldn't call beforeOperation when clearing the DataStore. The
         // only thing we have to wait for is the category initialization latch.
         boolean isCategoryInitialized = false;
         try {
-            isCategoryInitialized = categoryInitializationsPending.await(PLUGIN_INIT_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+            isCategoryInitialized = categoryInitializationsPending.await(LIFECYCLE_TIMEOUT_MS, TimeUnit.MILLISECONDS);
         } catch (InterruptedException exception) {
             LOG.warn("Execution interrupted while waiting for DataStore to be initialized.");
         }
         if (!isCategoryInitialized) {
```

Comment:
TLDR version: The DataStore.clear call may have to wait for the orchestrator to start. We might need to do something like this:

```suggestion
        if (!isCategoryInitialized || !orchestrator.isStarted()) {
```

If the clear API is called immediately after calling `Amplify.configure` but before the Orchestrator has a chance to fully start (i.e. it's in LOCAL_ONLY mode and in the process doing stuff before going into SYNC_VIA_API), the call to `DataStore.clear()` does the following:

- Call `orchestrator.stop()` (which is correct); however, the call to `transitionToStopped` only stops the storage observer because the current status of the orchestrator is LOCAL_ONLY. 
- Subscription attempt [is interrupted](https://github.com/aws-amplify/amplify-android/blob/da16e44a92d64f941fda4adb291cf88cd63c8d94/aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessor.java#L118) but the rest of the `startApiSync` continues because we're not re-throwing the interrupted exception.
- Orchestrator status is set to STOPPED.
- DataStore is cleared. (also correct)
- [Original call to `orchestrator.start()`] Somehow, the `startApiSync` call completes and sets the status SYNC_VIA_API
- Calls `orchestrator.start()` but this results in a no-op because the status is already SYNC_VIA_API. The problem is that at this point, the SubscriptionProcessor and MutationProcessor are in an unknown state. 

Using my sample app, the app does not crash but mutations are not being sent and subscriptions are not started.

Reply:
Thanks for this detailed analysis!

I started looking at additional steps that could be taken to tighten up the serialization guarantees in the Orchestrator. I think we can still do that.

But, in the interest of delivering this current fix, I will go with the `!orchestrator.isStarted()` approach you suggest. It isn't ideal, but as you note, we've solved most of the problems by now and are into the long tail.

Reply:
Totally agree with this. I think we can continue to iterate on this as needed, but I also feel that with that change, this PR should be considered ready to go.